testing/python/layout/test_tilelang_cute.py

```
@@ -0,0 +1,632 @@
+    assert cute.size(result) == cute.size(ref), f"size {cute.size(result)} != {cute.size(ref)}"
+    for i in range(cute.size(ref)):
+        assert result(i) == ref(i), f"at {i}: {result(i)} != {ref(i)}"
+
+
+def _build_swizzled_layout(shape, intermediate_fn, b_bits, m_base, s_shift):
+    """A single-output Layout whose forward index swizzles a bijective pow2
+    intermediate address, applying the swizzle exactly as SwizzleNode::Apply:
+    apply(x) = x ^ ((x & yyy) >> s_shift), with yyy = mask << (m_base + s_shift)
+    and mask = (1 << b_bits) - 1."""
+
+    def forward(*vars):
```

Ruff (0.15.17)

[error] Function argument `vars` is shadowing a Python builtin

(A002)